src/main/scala/net/jcazevedo/moultingyaml/ProductFormats.scala

```diff
@@ -240,8 +240,13 @@ trait ProductFormats {
           deserializationError(msg, cause, fieldName :: fieldNames)
       }
 
-    case _ => deserializationError("YamlObject expected in field '" + fieldName
-      + "'", fieldNames = fieldName :: Nil)
+    case YamlNull if isOption =>
+      None.asInstanceOf[A]
+
+    case _ =>
+      deserializationError((if (isOption) "Either YamlNull or " else "") +
```

Comment:
`YamlNull` only applies when the case class is wholly composed by `Option` fields. In other case classes, `YamlNull` is not valid and the message may mislead users.